aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/EventLoopGroupFactory.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import java.util.Optional;
+import java.util.concurrent.ThreadFactory;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Configurable factory for creating {@link EventLoopGroup} instances. Will choose the optimal implementation of
+ * {@link EventLoopGroup} per the platform (i.e. if on Linux then {@link EpollEventLoopGroup} will be used).
+ */
+public final class EventLoopGroupFactory
```

Comment:
Should we have an `EventLoopGroupFactory` interface with just a create on it so that customers can implement their own factories if they want to? Rather than this just being a `final` class.